pkg/cgroups/cgroups.go

```diff
@@ -127,20 +127,19 @@ func CheckOrMountCgrpFS(mapRoot string, runsOnKind bool) {
 		if mapRoot == "" {
 			mapRoot = cgroupRoot
 		}
-		err := cgrpCheckOrMountLocation(mapRoot)
-		// Failed cgroup2 mount is not a fatal error, sockmap will be disabled however
-		if err == nil {
-			log.Infof("Mounted cgroupv2 filesystem at %s", mapRoot)
-
-			if runsOnKind {
-				p, err := getCgroupRootForKind()
-				if err != nil {
-					log.WithError(err).
-						Warnf("Failed to determine cgroup v2 hierarchy for Kind node. Socket-based LB (--%s) will not work.",
-							option.EnableHostReachableServices)
-				} else {
-					setCgroupRoot(p)
-				}
+		if err := cgrpCheckOrMountLocation(mapRoot); err != nil {
+			log.WithError(err).Fatal("Failed to mount cgroup2")
```

Comment:
We need not fail early if socket-lb (the default kube-proxy disabled mode) is not enabled.
Also, we should probably mention why we have a fatal log. Something along these lines - "Failed to mount cgroup v2 filesystem needed by socket load balancing").

Reply:
> As far as I can tell, we never required cgroups v2 before, and most environments do not have cgroups v2 enabled, and this function is unconditionally run. This seems dangerous, if someone upgrades Cilium to this new version it will cause Cilium to crash and force the user to somehow resolve the issue. Am I missing something?

I clearly understand what you mean, @joestringer, and I understand the concern, but I wonder what @brb thinks based on his [note](https://github.com/cilium/cilium/pull/15999#issuecomment-848609493):

> I think we should not ignore the error. The comment in the code says:
>        // Failed cgroup2 mount is not a fatal error, sockmap will be disabled however
> We cannot no longer assume this, as bpf_sock (aka socket-lb) also depends on the mount.

I know how simple changes like these can demand a deeper analysis so, please, let us know how to help in this matter.